src/index.tsx

```diff
@@ -528,8 +526,48 @@ export async function validateServerFormData<T extends FormDefinition>(
   >;
   await Promise.all(
     entries.map(async ([inputName, inputDef]) => {
-      let values = formData.getAll(inputName);
-      if (formData.has(inputName)) {
+      if (!formData.has(inputName)) {
+        // No values submitted
+        let inputInfo: InputInfo = {
+          value: null,
+          touched: true,
+          dirty: true,
+          state: "done",
+          validity: await validateInput(
+            inputName,
+            inputDef.validationAttrs,
+            inputDef.customValidations,
+            "",
+            undefined,
+            formData
+          ),
+        };
+        inputs[inputName] = inputInfo;
+        // FIXME: ???
+        // @ts-expect-error
+        submittedValues[inputName] = null;
```

Comment:
Punted on this for now.  I'm happy to merge with these comments and come back to them later.  But it's not happy about us assigning specific values to the new complex type :/